### PR TITLE
Default system proxy not working with Account Wizard (fixes #2108)

### LINKED ISSUE #2108: [WIP] Default system proxy (PAC, Windows) not working with Account Wizard
### Expected behaviour
Tell us what should happen

### Actual behaviour
Tell us what happens instead

### Steps to reproduce
1.
2.
3.

### Client configuration
Client version:

Operating system:

OS language:

Qt version used by client package (Linux only, see also Settings dialog):

Client package (From Nextcloud or distro) (Linux only):

Installation path of client:

### Server configuration

Nextcloud version:

Storage backend (external storage):

### Logs

Please use Gist (https://gist.github.com/) or a similar code paster for longer
logs.

1. Client logfile: Output of `nextcloud --logwindow` or `nextcloud --logfile log.txt`
(On Windows using `cmd.exe`, you might need to first `cd` into the Nextcloud directory)
(See also https://docs.nextcloud.com/desktop/2.3/troubleshooting.html#log-files)

2. Web server error log:

3. Server logfile: nextcloud log (data/nextcloud.log):

### PR DESCRIPTION
**Pull request by misch7**
Discovered on Windows in conjunction with `PAC` scripts:
- Already configured accounts worked
- Fresh client account setup did not work

Reason:
- Proxy was reset over and over again in `Account::setCredentials`

### CODE REVIEW

File: src/libsync/account.cpp [hunk 1]
```diff
@@ -181,6 +189,9 @@ void Account::setCredentials(AbstractCredentials *cred)
     if (jar) {
         _am->setCookieJar(jar);
     }
+    if (hadProxy) {
```
Comment (er-vin) [inline src/libsync/account.cpp:192]: isn't `hadProxy` a proxy value (ah!) for `proxy.type() != QNetworkProxy::NoProxy`?

Comment (misch7) [reply src/libsync/account.cpp]: actually `QNetworkProxy::DefaultProxy` after some digging and testing ;)

**Comment by misch7** [comment]
> I'm wondering if we wouldn't need the same type of fix in `resetNetworkAccessManager()` as well. After all we go through the trouble of always saving the cookie jar and it looks like the proxy should get the same treatment.

Hehe funny, I thought the same and did it before I read the comment ^^

**Comment by misch7** [comment]
🍒 Cherry-picked to stable-2.6

**Pull request merged**